### DISCUSSION TITLE
Show link in logbook for script trace

### DISCUSSION THREAD

**Discussion by aneisch** [home-assistant/frontend#9605] [Other feature requests]
Currently in the logbook there is a link under an automation entry to show the automation trace.

![Screen Shot 2021-07-26 at 1 17 15 PM](https://user-images.githubusercontent.com/6991461/127039686-0e3cf503-aa14-45b2-a591-18ed57530b72.png)

Now that scrips also have traces available, it would be great to have a link to the trace for the script execution in the logbook. Currently I believe the only way to view scripts traces is from the scripts dashboard.

**Comment by spacegaier** [comment]
I like the idea. Two things we need to keep in mind though:
1. Logbook entries are persisted across system restarts, where as traces I think are removed once HA is shutdown or restarted. So we would need to check if a trace exists. Not sure how performance intensive that check would be.
2. Similar to the point above, we need to find the correct trace, since for an automation or script there can be multiple.

@matthiasdebaat What do you think UX wise?

**Reply by matthiasdebaat** [reply]
Great idea to make them consistent, also like the smal link. In terms of performance, could this affect HA booting or loading Logbook?

**Reply by spacegaier** [reply]
Your comment just made me realize that we already have that "Show trace" link in the logbook, but only for automation (apparently did not read OP's post carefully enough 🙈 ). I thought that is something completely new (had never seen that link in my HA instances so far).

Re performance: Given that we already do it today for automations, I don't think doing the same for scripts will have a big impact...unless somehow runs a tons of scripts, but I think users have more automations compared to scripts.

**Reply by spacegaier** [reply]
I had a quick first look, but it seems the backend does not yet provide the `context_id` for "script_started" events. So not an easy frontend change.

**Comment by drinfernoo** [comment]
I think it'd be really nice to have a link to the traces in the "More Info" dialog for an automation/script :wink:
